eng/scripts/mark-shipped.ps1

```diff
@@ -0,0 +1,46 @@
+[CmdletBinding(PositionalBinding=$false)]
+param ()
+
+Set-StrictMode -version 2.0
+$ErrorActionPreference = "Stop"
+
+function MarkShipped([string]$dir) {
+    $shippedFilePath = Join-Path $dir "PublicAPI.Shipped.txt"
+    $shipped = Get-Content $shippedFilePath
+    if ($null -eq $shipped) {
+        $shipped = @()
+    }
+
+    $unshippedFilePath = Join-Path $dir "PublicAPI.Unshipped.txt"
+    $unshipped = Get-Content $unshippedFilePath
+    $removed = @()
+    $removedPrefix = "*REMOVED*";
+    Write-Host "Processing $dir"
+
+    foreach ($item in $unshipped) {
+        if ($item.Length -gt 0) {
+            if ($item.StartsWith($removedPrefix)) {
+                $item = $item.Substring($removedPrefix.Length)
+                $removed += $item
+            }
+            else {
+                $shipped += $item
+            }
+        }
+    }
+
+    $shipped | Sort-Object -Unique |Where-Object { -not $removed.Contains($_) } | Out-File $shippedFilePath -Encoding Ascii
+    Copy-Item eng/PublicAPI.empty.txt $unshippedFilePath
```

Comment:
/fyi @tmat we needed the `-Unique` to avoid duplicating `#nullable enable` lines in PublicAPI.Shipped.txt files and the `Copy-Item` to avoid _removing_ those lines from the PublicAPI.Unshipped.txt files.